src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateDublicate = { ...state };
+  const result = [];
+
+  for (const property of actions) {
+    switch (property.type) {
+      case 'addProperties':
+        Object.assign(stateDublicate, property.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const names of property.keysToRemove) {
+          delete stateDublicate[names];
+        }
+        break;
+
+      case 'clear':
+        for (const allProprety in stateDublicate) {
+          delete stateDublicate[allProprety];
+        }
```

Comment:
`default` block in `switch` statement is required, so you should add it. It can prevent some errors